sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SplitSkillEncoderModelName.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// 
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package com.azure.search.documents.indexes.models;
+
+import com.azure.core.util.ExpandableStringEnum;
+import java.util.Collection;
+
+/**
+ * A value indicating which tokenizer to use.
+ */
+public final class SplitSkillEncoderModelName extends ExpandableStringEnum<SplitSkillEncoderModelName> {
+    /**
+     * Refers to a base model trained with a 50,000 token vocabulary, often used in general natural language processing
+     * tasks.
+     */
+    public static final SplitSkillEncoderModelName R50KBASE = fromString("r50k_base");
```

Comment:
We're going to get feedback on the naming here (I know this is what Autorest generates). We'll likely need customizations to turn this into `R_50K_BASE`/.